src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -74,19 +82,40 @@ public function count()
      */
     public function add($name, Route $route)
     {
-        unset($this->routes[$name]);
+        $this->addWithPriority($name, $route, 0);
+    }
 
-        $this->routes[$name] = $route;
+    /**
+     * Adds a route.
+     *
+     * @param string $name  The route name
+     * @param Route  $route A Route instance
+     */
+    public function addWithPriority($name, Route $route, int $priority = 0)
```

Comment:
should be removed and a third arg added to `add()`, using the common technique we use elsewhere (docblock `@param` + commented argument in `/* ...*/` + `func_num/get_arg(s)`